manifest.go

```diff
@@ -324,6 +325,71 @@ func ValidateProjectRoots(c *Ctx, m *Manifest, sm gps.SourceManager) error {
 	return valErr
 }
 
+// ValidatePackageRules ensure that there are no ineffectual package declarations
+// in the constraints, overrides, and ignored rules in the manifest.
+func ValidatePackageRules(c *Ctx, proj *Project, sm gps.SourceManager) (err error) {
+	directDeps, err := proj.DirectDependencies(sm)
+	if err != nil {
+		return err
+	}
+
+	ic := FindIneffectualConstraints(proj.Manifest, directDeps)
+	if len(ic) > 0 {
+		err = errInefficientRules
+		c.Err.Printf("The following constraints aren't imported:\n\n")
+		for _, pr := range ic {
+			c.Err.Println("  ", pr)
+		}
+		c.Err.Println()
+	}
+
+	ig := FindIneffectualIgnores(proj.Manifest, directDeps)
+	if len(ig) > 0 {
+		err = errInefficientRules
+		c.Err.Printf("The following ignored packages aren't imported:\n\n")
+		for _, pr := range ig {
+			c.Err.Println("  ", pr)
+		}
+		c.Err.Println()
+	}
+
+	return err
+}
+
+// FindIneffectualConstraints looks for constraints decleared in the project
+// manifest that aren't directly used in the project.
+func FindIneffectualConstraints(m *Manifest, directDeps map[string]bool) []gps.ProjectRoot {
```

Comment:
It looks like the `FindIneffectual*` functions don't *need* to be exported, is there a compelling reason to export them?

Reply:
They don't need to be, I'll fix that